src/Draco.Lsp/Server/LanguageClientProxy.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Draco.Lsp.Attributes;
+
+
+namespace Draco.Lsp.Server;
+
+/// <summary>
+/// Intercepts language client calls.
+/// </summary>
+internal class LanguageClientProxy : DispatchProxy
```

Comment:
Can this be sealed?

Reply:
No, `DispatchProxy` generates a class that derives from this at runtime